src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -81,13 +81,19 @@ private static ITextAndVersionSource CreateRecoverableText(TextAndVersion text,
         {
             var result = new RecoverableTextAndVersion(new ConstantTextAndVersionSource(text), services);
 
-            // This RecoverableTextAndVersion is created directly from a TextAndVersion instance. In its initial state,
-            // the RecoverableTextAndVersion keeps a strong reference to the initial TextAndVersion, and only
-            // transitions to a weak reference backed by temporary storage after the first time GetValue (or
-            // GetValueAsync) is called. Since we know we are creating a RecoverableTextAndVersion for the purpose of
-            // avoiding problematic address space overhead, we call GetValue immediately to force the object to weakly
-            // hold its data from the start.
-            result.GetValue(loadTextOptions, CancellationToken.None);
+            var service = services.GetRequiredService<IWorkspaceConfigurationService>();
+            var options = service.Options;
+
+            if (!options.DeferCreatingRecoverableText)
+            {
+                // This RecoverableTextAndVersion is created directly from a TextAndVersion instance. In its initial state,
+                // the RecoverableTextAndVersion keeps a strong reference to the initial TextAndVersion, and only
+                // transitions to a weak reference backed by temporary storage after the first time GetValue (or
+                // GetValueAsync) is called. Since we know we are creating a RecoverableTextAndVersion for the purpose of
+                // avoiding problematic address space overhead, we call GetValue immediately to force the object to weakly
+                // hold its data from the start.
+                result.GetValue(loadTextOptions, CancellationToken.None);
```

Comment:
i'd like to remove this code (esp. how it grabs locks with CT.None, and is otherwise doing potentially expensive synchronous work in an uncancellable fashion).  Adding things as a workspace config option so we can A/B test this and see if there are any regressions we see.